### PR TITLE
TE-4789 - Implement TLS confugiratiuon for RMQ

### PR DESCRIPTION
**Pull request by jacoz**
- Developer(s): @jacoz

- Ticket: https://spryker.atlassian.net/browse/TE-4788

- Release Group: https://release.spryker.com/release-groups/view/2124

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   RabbitMq              | minor                 |                       |

#### Release Notes

{skip}

-----------------------------------------

##### Change log

Improvements
- Added the possibility to configure stream context options for secure connections.
- Introduced `RabbitMqEnv::RABBITMQ_STREAM_CONTEXT_OPTIONS` config.
- Introduced `QueueConnectionTransfer.streamContextOptions` transfer property.

### CODE REVIEW

File: src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php [hunk 1]
```diff
@@ -114,14 +114,35 @@ protected function createAmqpStreamConnection(QueueConnectionTransfer $queueConn
...
             $queueConnectionTransfer->getSslProtocol()
         );
     }
 
+    /**
+     * @param \Generated\Shared\Transfer\QueueConnectionTransfer $queueConnectionTransfer
+     *
+     * @return resource|null
+     */
+    private function createSslContext(QueueConnectionTransfer $queueConnectionTransfer)
```
Comment (gechetspr) [inline src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php]: why this is private? We can make it protected if it's not required in this case to be private

File: src/Spryker/Shared/RabbitMq/Transfer/rabbit_mq.transfer.xml [hunk 2]
```diff
@@ -23,6 +23,7 @@
         <property name="queueOptionCollection" singular="queueOptionItem" type="RabbitMqOption[]"/>
         <property name="isDefaultConnection" type="bool"/>
         <property name="storeNames" type="array" singular="storeNames"/>
+        <property name="streamContextOptions" type="array" singular="streamContextOptions"/>
```
Comment (gechetspr) [inline src/Spryker/Shared/RabbitMq/Transfer/rabbit_mq.transfer.xml]: as I know, for the new fields singular field, should be singular :)
We have it in plural for old cases to keep BC

File: src/Spryker/Shared/RabbitMq/RabbitMqEnv.php [hunk 3]
```diff
@@ -137,4 +137,12 @@ interface RabbitMqEnv extends QueueConstants
      * @api
      */
     public const RABBITMQ_EXCHANGE_BLACKLIST = 'RABBITMQ:RABBITMQ_EXCHANGE_BLACKLIST';
+
+    /**
+     * Specification:
+     * - Use this constant to configure the stream context
```
Comment (gechetspr) [inline src/Spryker/Shared/RabbitMq/RabbitMqEnv.php]: Please add a dot in the end.

File: src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php [hunk 4]
```diff
@@ -114,14 +114,35 @@ protected function createAmqpStreamConnection(QueueConnectionTransfer $queueConn
             $queueConnectionTransfer->getLocale() ?? $this->getDefaultLocale(),
             $queueConnectionTransfer->getConnectionTimeout() ?? $defaultQueueConnectionTransfer->getConnectionTimeout(),
             $queueConnectionTransfer->getReadWriteTimeout() ?? $defaultQueueConnectionTransfer->getReadWriteTimeout(),
-            null,
+            $this->createSslContext($queueConnectionTransfer),
```
Comment (gechetspr) [inline src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php]: I would expect to have more generic context here. Config value doesn't say anything that it's `ssl` only related

File: src/Spryker/Shared/RabbitMq/Transfer/rabbit_mq.transfer.xml (hunk 2 above)
Comment (jacoz) [reply src/Spryker/Shared/RabbitMq/Transfer/rabbit_mq.transfer.xml]: Uhh okay, I just used the same approach, I'll make it singular.

File: src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php (hunk 1 above)
Comment (jacoz) [reply src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php]: I thought private was better in this scenario cause this logic shouldn't be overwritten.

But if you see a scenario where it actually could, I can make it protected.

File: src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php (hunk 4 above)
Comment (jacoz) [reply src/Spryker/Client/RabbitMq/Model/Connection/ConnectionBuilder/ConnectionBuilder.php]: True, I'll rename it.

**Pull request merged**